extensions/wikia/WikiFactory/SpecialWikiFactory_body.php

```diff
@@ -61,160 +65,201 @@ public function execute( $subpage ) {
-		$tab = "variables";
-		if( is_null( $cityid ) && ( isset( $subpage ) || isset( $domain ) ) ) {
+		return WikiFactory::getWikiByID( $cityId );
+	}
 
-			/**
-			 * if there is # in subpage we are switching tab
-			 */
-			if( strpos( $subpage, "/" ) ) {
-				$parts = explode( "/", $subpage, 3 );
-				if( is_array( $parts ) && sizeof( $parts ) >= 2 ) {
-					$subpage = $parts[0];
-					$tab = $parts[1];
-					if ( ( $tab === "variables" ) && ( isset($parts[2]) ) ) {
-						$this->mVariableName = trim($parts[2]);
-					}
-					if ( ( $tab === "tags" || $tab === "findtags"  ) && ( isset($parts[2]) ) ) {
-						$this->mSearchTag = trim($parts[2]);
-					}
-				}
-			}
+	/**
+	 * Take care of setting internal variables necessary to handle the different
+	 * navigation tabs we could be on
+	 *
+	 * @param string $subpage
+	 */
+	private function handleNavigationTab( $subpage ) {
+		// The default
+		$this->mTab = "variables";
+		$this->mVariableName = "";
 
-			if( is_null( $domain ) ) {
-				$domain = $subpage;
-			}
+		// If there are no parts after the wiki ID, there's no tab to switch to
+		if ( !strpos( $subpage, "/" ) ) {
+			return;
+		}
 
-			if( ctype_digit( $subpage ) ) {
-				$cityid = $subpage;
+		// if there is # in subpage we are switching tabs
+		$parts = explode( "/", $subpage, 3 );
+		if ( is_array( $parts ) && sizeof( $parts ) >= 2 ) {
+			$this->mTab = $parts[1];
+			if ( ( $this->mTab === "variables" ) && ( isset( $parts[2] ) ) ) {
+				$this->mVariableName = trim( $parts[2] );
 			}
-			else {
-				/**
-				 * if $domain starts with db: it means that we want to search by database
-				 * if $domain starts with id: it means that we want to search by city_id
-				 */
-				$cityid = null;
-				if( preg_match( "/^db\:/", $domain ) ) {
-					list( $prefix, $name ) = explode( ":", $domain, 2 );
-					$cityid = WikiFactory::DBtoID( strtolower( $name ) );
-				}
-				elseif( preg_match( "/^id\:/", $domain ) ) {
-					list( $prefix, $cityid ) = explode( ":", $domain, 2 );
-				}
-				else {
-					/**
-					 * if there's no dot in cityname we add .wikia.com
-					 * or if is only one dot (language.domain.wikia.com)
-					 */
-					if( sizeof(explode(".", $domain )) <= 2 && strlen( $domain ) > 0) {
-						$this->mDomain = $domain;
-						$domain = $domain.".".$wgWikiaBaseDomain;
-					} else {
-						list( $code, $subdomain ) = explode(".", $domain, 2 );
-						$exists = 0;
-						if ( in_array( $code, array_keys( Language::getLanguageNames() ) ) ) {
-							$exists = 1;
-						} else {
-							$cityid = WikiFactory::DomainToId( $domain );
-							if ( !is_null($cityid) ) {
-								$exists = 1;
-							}
-						}
-						if ( $exists == 1 ) {
-							$this->mDomain = $code.".".$subdomain;
-							$domain = sprintf("%s.%s", $code, $subdomain );
-						}
-					}
-					if ( is_null($cityid) ) {
-						$cityid = WikiFactory::DomainToId( $domain );
-					}
-				}
+			if ( ( $this->mTab === "tags" || $this->mTab === "findtags" )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "findtags" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)